src/action.ts

```diff
@@ -116,8 +130,141 @@ async function setupBinaryInEnv(binary: DownloadedBinary) {
   await exec.exec("chmod", ["+x", binaryPath]);
 }
 
-async function downloadGitHubBinary(
-  octokit: GitHub,
+class GitHubProvider implements ScmProvider {
+  private octokit: GitHub;
+
+  constructor(config: ScmConfig) {
+    this.octokit = github.getOctokit(config.token, {
+      baseUrl:
+        config.baseUrl === "https://github.com"
+          ? "https://api.github.com"
+          : `${config.baseUrl}/api/${config.apiVersion}`,
+    });
+  }
+
+  async getReleaseByTag(owner: string, repo: string, tag: string): Promise<any> {
```

Comment:
`Promise<any>` is a very loose interface, we should define an actual return type here so calling code down below is guaranteed correct